ledger/metrics/src/main/scala/com/daml/metrics/Metrics.scala

```diff
@@ -39,15 +39,15 @@ final class Metrics(val registry: MetricRegistry) {
         registry.meter(Prefix :+ "valid_submissions")
 
       def inputBufferLength(party: String): Counter =
-        registry.counter(Prefix :+ party :+ "input_buffer_length")
+        registry.counter(Prefix :+ party.substring(0, 64) :+ "input_buffer_length")
```

Comment:
While truncation is of the solution, the other half must be not to use concatenated party names to name the metrics.
Parties on behalf of whom a submission happens can come in many different combinations and we are risking a combinatoric explosion of the metric namespace.

Reply:
Updated to just use first (sorted) party in metrics name